src/main/java/com/minecolonies/coremod/entity/ai/citizen/healer/EntityAIWorkHealer.java

```diff
@@ -0,0 +1,527 @@
+                }
+            }
+
+            if (patient.getState() == Patient.PatientState.TREATED)
+            {
+                if (disease == null)
+                {
+                    this.currentPatient = patient;
+                    return CURE;
+                }
+
+                if (!hasCureInInventory(disease, citizen.getInventoryCitizen()))
+                {
+                    patient.setState(Patient.PatientState.NEW);
+                    return DECIDE;
+                }
+            }
+        }
+
+        for (final PlayerEntity player : world.getEntitiesWithinAABB(EntityType.PLAYER, getTargetableArea(), player -> player.getHealth() < player.getMaxHealth() - 10 -  (2 * getOwnBuilding().getBuildingLevel())))
+        {
+            playerToHeal = player;
+            return CURE_PLAYER;
+        }
+
+        final ICitizenData data = getOwnBuilding().getColony().getCitizenManager().getRandomCitizen();
+        if (data.getCitizenEntity().isPresent() && data.getCitizenEntity().get().getHealth() < 10.0 && BlockPosUtil.getDistance2D(data.getCitizenEntity().get().getPosition(), getOwnBuilding().getPosition()) < getOwnBuilding().getBuildingLevel() * 40)
+        {
+            remotePatient = data;
+            return WANDER;
+        }
+        return DECIDE;
+    }
+
+
+    /**
+     * Request the cure for a given patient.
+     * @return the next state to go to.
+     */
+    private IAIState requestCure()
+    {
+        if (currentPatient == null)
+        {
+            return DECIDE;
+        }
+
+        final ICitizenData data = getOwnBuilding().getColony().getCitizenManager().getCitizen(currentPatient.getId());
+        if (data == null || !data.getCitizenEntity().isPresent() || !data.getCitizenEntity().get().getCitizenDiseaseHandler().isSick())
+        {
+            currentPatient = null;
+            return DECIDE;
+        }
+
+        final EntityCitizen citizen = (EntityCitizen) data.getCitizenEntity().get();
+        if (walkToBlock(citizen.getPosition()))
+        {
+            return REQUEST_CURE;
+        }
+
+        final String diseaseName = citizen.getCitizenDiseaseHandler().getDisease();
+        if (diseaseName.isEmpty())
+        {
+            currentPatient.setState(Patient.PatientState.REQUESTED);
+            currentPatient = null;
+            return DECIDE;
+        }
+
+        final ImmutableList<IRequest<? extends Stack>> list = getOwnBuilding().getOpenRequestsOfType(worker.getCitizenData(), TypeToken.of(Stack.class));
+        for (final ItemStack cure : IColonyManager.getInstance().getCompatibilityManager().getDisease(diseaseName).getCure())
+        {
+            boolean hasRequest = false;
+            for (final IRequest<? extends Stack> request : list)
+            {
+                if (request.getRequest().getStack().isItemEqual(cure))
+                {
```

Comment:
same as above, this probably doesnt work for requesting the same cure twice

Reply:
In this regard, it doesn't matter much. The healer will re-request it then later on. After he notices he is missing a cure for another citizen.